### PR TITLE
Cookies endpoint

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: endpoints/delete_cookies.md [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+## Response
+
+```json
+{
+	"results": [
+      {
+        "account_id": "acct_1234",
+        "cookie_id": "ck_1234",
+        "website_id": "web_1234",
+        "_idx": 1
```
Comment (josh-termly) [inline endpoints/delete_cookies.md]: `_idx` should be 0

File: endpoints/delete_cookies.md [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
```
Comment (josh-termly) [inline endpoints/delete_cookies.md]: `_idx` should be 0

File: endpoints/post_cookies.md [hunk 3]
```diff
@@ -0,0 +1,347 @@
+# Overview
+
+Creates cookies. The requst body has the following shape:
+
+```json
+{
+  "account_id": "<string>",
+  "category": "<string>",
```
Comment (josh-termly) [inline endpoints/post_cookies.md]: I believe this is an enum.

File: endpoints/post_cookies.md [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+  "expire": "<string>",
+  "fr": "<string>",
+  "initiator": "<string>",
+  "name": "<string>",
+  "new": "<bool>",
+  "party_provider_type": "<string>",
+  "service": "<string>",
+  "service_policy_link": "<string>",
+  "source": "<string>",
+  "tracker_type": "<string>",
```
Comment (josh-termly) [inline endpoints/post_cookies.md]: I believe this is an enum

File: endpoints/post_cookies.md [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+```
+
+The results will contain zero or more of the following objects:
+
+```json
+{
+  "account_id": "<string>",
+  "cookie_id": "<string>",
+  "website_id": "<string>",
+  "_idx": "<number>"
```
Comment (josh-termly) [inline endpoints/post_cookies.md]: `<number>` or `<integer>`?

File: endpoints/post_cookies.md [hunk 6]
```diff
@@ -0,0 +1,347 @@
...
+## Response
+
+```json
+{
+	"results": [
+      {
+        "account_id": "acct_1234",
+        "cookie_id": "ck_1234",
+        "website_id": "web_1234",
+        "_idx": 1
```
Comment (josh-termly) [inline endpoints/post_cookies.md:120]: 0 not 1?

File: endpoints/post_cookies.md [hunk 7]
```diff
@@ -0,0 +1,347 @@
...
```
Comment (josh-termly) [inline endpoints/post_cookies.md:226]: zero index.

File: endpoints/post_cookies.md [hunk 8]
```diff
@@ -0,0 +1,347 @@
...
```
Comment (josh-termly) [inline endpoints/post_cookies.md:322]: 0

File: endpoints/put_cookies.md [hunk 9]
```diff
@@ -0,0 +1,355 @@
+# Overview
+
+Updates cookies. The request body has the following shape:
+
+```json
+{
+  "account_id": "<string>",
+  "category": "<string>",
```
Comment (josh-termly) [inline endpoints/put_cookies.md]: I believe enum

File: endpoints/put_cookies.md [hunk 10]
```diff
@@ -0,0 +1,355 @@
...
+  "fr": "<string>",
+  "id": "<string>",
+  "initiator": "<string>",
+  "name": "<string>",
+  "new": "<bool>",
+  "party_provider_type": "<string>",
+  "service": "<string>",
+  "service_policy_link": "<string>",
+  "source": "<string>",
+  "tracker_type": "<string>",
```
Comment (josh-termly) [inline endpoints/put_cookies.md]: I believe enum

File: endpoints/put_cookies.md [hunk 11]
```diff
@@ -0,0 +1,355 @@
...
+```
+
+The results will contain zero or more of the following objects:
+
+```json
+{
+  "account_id": "<string>",
+  "cookie_id": "<string>",
+  "website_id": "<string>",
+  "_idx": "<number>"
```
Comment (josh-termly) [inline endpoints/put_cookies.md:72]: as previous `<number>` or `<integer>`

File: endpoints/put_cookies.md [hunk 12]
```diff
@@ -0,0 +1,355 @@
...
+## Response
+
+```json
+{
+	"results": [
+      {
+        "account_id": "acct_1234",
+        "cookie_id": "ck_1234",
+        "website_id": "web_1234",
+        "_idx": 1
```
Comment (josh-termly) [inline endpoints/put_cookies.md:123]: zero indexed

File: endpoints/put_cookies.md [hunk 13]
```diff
@@ -0,0 +1,355 @@
...
```
Comment (josh-termly) [inline endpoints/put_cookies.md:232]: zero indexed

File: endpoints/put_cookies.md [hunk 14]
```diff
@@ -0,0 +1,355 @@
...
```
Comment (josh-termly) [inline endpoints/put_cookies.md:330]: zero indexed

File: endpoints/delete_cookies.md [hunk 15]
```diff
@@ -0,0 +1,195 @@
...
+    "id": "ck_1234",
+    "website_id": "web_1234"
+  }
+]
+```
+
+## Response
+
+```json
+{
```
Comment (josh-termly) [inline endpoints/delete_cookies.md:64]: I believe this is the get response

File: endpoints/delete_cookies.md [hunk 16]
```diff
@@ -0,0 +1,195 @@
...
+    "id": "ck_0123",
+    "website_id": "web_1234"
+  }
+]
+```
+
+## Response
+
+```json
+{
```
Comment (josh-termly) [inline endpoints/delete_cookies.md:114]: I believe this is the get response

File: endpoints/delete_cookies.md [hunk 17]
```diff
@@ -0,0 +1,195 @@
...
+	"results": [
```
Comment (josh-termly) [inline endpoints/delete_cookies.md:171]: I believe this is the get response

File: endpoints/post_cookies.md [hunk 18]
```diff
@@ -0,0 +1,347 @@
...
+Creates a single cookie for a single website.
+
+## Request
+
+POST `/api/v3/websites/cookies`
+
+## Query
+
+```json
+{
```
Comment (josh-termly) [inline endpoints/post_cookies.md]: I think we decided to go with the array being the top level

File: endpoints/post_cookies.md [hunk 19]
```diff
@@ -0,0 +1,347 @@
...
+            "website_id": "web_4567"
+        }
+    ]
+}
+```
+
+## Response
+
+```json
+{
```
Comment (josh-termly) [inline endpoints/post_cookies.md:220]: I believe this is the get response

File: endpoints/post_cookies.md [hunk 20]
```diff
@@ -0,0 +1,347 @@
...
+            "website_id": "web_0987"
+        }
+    ]
+}
+```
+
+## Response
+
+```json
+{
```
Comment (josh-termly) [inline endpoints/post_cookies.md:316]: I believe this is the get response

File: endpoints/put_cookies.md [hunk 21]
```diff
@@ -0,0 +1,355 @@
...
+        }
+    ]
+}
+```
+
+## Response
+
+```json
+{
+	"results": [
```
Comment (josh-termly) [inline endpoints/put_cookies.md:118]: I believe this is the get response

File: endpoints/put_cookies.md [hunk 22]
```diff
@@ -0,0 +1,355 @@
...
```
Comment (josh-termly) [inline endpoints/put_cookies.md:227]: I believe this is the get response

File: endpoints/put_cookies.md [hunk 23]
```diff
@@ -0,0 +1,355 @@
...
+            "website_id": "web_0987"
+        }
+    ]
+}
+```
+
+## Response
+
+```json
+{
```
Comment (josh-termly) [inline endpoints/put_cookies.md:324]: I believe this is the get response

File: endpoints/post_cookies.md [hunk 24]
```diff
@@ -0,0 +1,347 @@
...
+
+## Request
+
+POST `/api/v3/websites/cookies`
+
+## Query
+
+```json
+{
+    "items": [
```
Comment (josh-termly) [inline endpoints/post_cookies.md]: I think we decided to go with the array being the top level

File: endpoints/post_cookies.md [hunk 25]
```diff
@@ -0,0 +1,347 @@
...
```
Comment (josh-termly) [inline endpoints/post_cookies.md]: I think we decided to go with the array being the top level

File: endpoints/put_cookies.md [hunk 26]
```diff
@@ -0,0 +1,355 @@
...
+
+## Request
+
+PUT `/api/v3/websites/cookies`
+
+## Query
+
+```json
+{
+    "items": [
```
Comment (josh-termly) [inline endpoints/put_cookies.md]: I think we decided to go with the array being the top level

File: endpoints/put_cookies.md [hunk 27]
```diff
@@ -0,0 +1,355 @@
...
```
Comment (josh-termly) [inline endpoints/put_cookies.md]: I think we decided to go with the array being the top level

File: endpoints/put_cookies.md [hunk 28]
```diff
@@ -0,0 +1,355 @@
...
```
Comment (josh-termly) [inline endpoints/put_cookies.md]: I think we decided to go with the array being the top level

File: endpoints/read_cookies.md [hunk 29]
```diff
@@ -0,0 +1,293 @@
...
+	"paging": {}
+}
+```
+
+The results will contain zero or more of the following objects:
+
+```json
+{
+  "account_id": "<string>",
+  "category": "<string>",
```
Comment (josh-termly) [inline endpoints/read_cookies.md]: enum

File: endpoints/read_cookies.md [hunk 30]
```diff
@@ -0,0 +1,293 @@
...
+  "fr": "<string>",
+  "id": "<string>",
+  "initiator": "<string>",
+  "name": "<string>",
+  "new": "<bool>",
+  "party_provider_type": "<string>",
+  "service": "<string>",
+  "service_policy_link": "<string>",
+  "source": "<string>",
+  "tracker_type": "<string>",
```
Comment (josh-termly) [inline endpoints/read_cookies.md]: enum

**Review by josh-termly — CHANGES_REQUESTED** [review]

File: endpoints/post_cookies.md (hunk 5 above)
Comment (ghost) [reply endpoints/post_cookies.md]: *Technically* JSON -- due to its JavaScript progenitor -- uses a single representation for numbers; number. I am fine with using integer if that's what we all say.

Comment (ghost) [reply endpoints/post_cookies.md]: changed to integer

**Review by josh-termly — APPROVED** [review]

**Pull request merged**